interface/bigTree.h

```diff
@@ -1110,4 +1131,43 @@ class bigTree {
   }
 };
 
+class correctedLeptons { 
+public:
+  unsigned N = 0;
+  std::vector<float> px;
+  std::vector<float> py;
+  std::vector<float> pz;
+  std::vector<float> en;
+
+  correctedLeptons (const unsigned N_, bigTree* theBigTree) {
+	N = N_;
+	px.reserve(N);
+	py.reserve(N);
+	pz.reserve(N);
+	en.reserve(N);
+
+	correct(theBigTree);
+  };
+
+  void correct(bigTree* theBigTree) {
+	for (unsigned iLep=0; iLep<N; ++iLep)
+	  {
+		TLorentzVector tlv_corr(theBigTree->daughters_px->at(iLep), theBigTree->daughters_py->at(iLep),
+								theBigTree->daughters_pz->at(iLep), theBigTree->daughters_e->at(iLep));
+		double uncorrEn = tlv_corr.E();
+		if(theBigTree->particleType->at(iLep) == 1) {
```

Comment:
for my own education: sometime electrons are checked with `particleType`, sometimes with `GenMatch`; any reason why?

Reply:
```particleType``` gives you the lepton type at RECO level (I believe): electron, muon or tau. ```genmatch``` gives you how the lepton was generated relative to a tau i.e., if it is prompt or if it comes from a tau decay.